### PR TITLE
feat: add user crud

### PR DESCRIPTION
**Pull request by JheremyTancara**
**Description:**
Implemented full CRUD operations for the User module, including creating, reading, updating, and deleting users. Added data validations and structured JSON responses for easy integration.

### CODE REVIEW

**Comment by GastonGutierrezC** [comment]
Add the communication with the API gateway and make it work through the API gateway.

**Comment by GastonGutierrezC** [comment]
Add a better description of what specific enpoints you have made and what is the purpose of each one of them.

**Comment by JheremyTancara** [comment]
> Add a better description of what specific enpoints you have made and what is the purpose of each one of them.

The description is already clearly written, each controller is located on the endpoint route.
E.g.:
* For UsersController it is 'users'

**Comment by JheremyTancara** [comment]
> Add the communication with the API gateway and make it work through the API gateway.

The connection to the API gateway is now established.

File: apps/auth-service/src/app.module.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+  imports: [
+    UsersModule,
+    UsersInformationsModule,
+    TypeOrmModule.forRoot({
+      type: "mongodb",
+      host: "auth-db",
+      port: 27017,
+      username: "auth",
+      password: "auth",
+      database: "auth-db",
```
Comment (AndrePzDn) [inline apps/auth-service/src/app.module.ts:16]: This info should be in a `.env` file

**Review by AndrePzDn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**